molecule/src/molecule_name_parser.cpp

```diff
@@ -531,8 +531,18 @@ void MoleculeNameParser::FragmentNodeBase::print(ostream& out) const
     out << "Type: FragmentNodeBase" << endl;
 
     out << "Multipliers:" << endl;
-    const auto& mul_container = multipliers._Get_container();
-    std::for_each(mul_container.begin(), mul_container.end(), [&out](const Multiplier& multiplier) { out << "\tvalue: " << multiplier.first << endl; });
+    auto muls = multipliers;
+    Multipliers muls_rev;
+    while (!muls.empty())
+    {
+        muls_rev.push(muls.top());
+        muls.pop();
+    }
+    while (!muls_rev.empty())
+    {
+        out << "\tvalue: " << muls_rev.top().first << endl;
+        muls_rev.pop();
+    }
```

Comment:
Maybe better change the container type of Multipliers, than transfuse the stack twice, not counting the initial copy-in-assignment.

Reply:
Maybe better but refactoring of molecule_name_parser.cpp is a separate task.

I've patched compilation error for VS2019 (const auto& mul_container = multipliers._Get_container();<=_Get_container is deprecated) with minimal code intervention. If you want something more, please create an issue and assign it to me.

Reply:
OK. will create an issue for optimization. Closing this.

Reply:
Oh, just realized that this change is a fix for `_Get_container` failed compilation in latest Visual Studio and that was hidden under `#ifdef DEBUG` . So it didn't harm release builds.
I hope this change is quite isolated from other changes in this pull-request, right?
How about the following: make a separate pull-request for this fix alone and also try to remove `#ifdef DEBUG`? So the class (`FragmentNodeSubstituent`) would be the same for all platforms and configurations.

Reply:
The separate PR: https://github.com/epam/Indigo/pull/294

This is a debugging output only, don't worry. Prod is not affected.